### PR TITLE
add content_encoding to headers_out

### PR DESCRIPTION
**Pull request by crowell**
this change is required for gzip cache change going into mod_pagespeed.

### CODE REVIEW

**Comment by crowell** [comment]
@jeffkaufman this change unblocks the gzipped-cache change and allows it to go it.

**Comment by jeffkaufman** [comment]
The shortcuts in `headers_out_t` are:
```
server, date, content_length, content_encoding, location, refresh, last_modified,
content_range, accept_ranges, www_authenticate, expires, and etag
```

With this change we'll be populating:
```
Date, Etag, Expires, Last-Modified, Location, Server,
Content-Length, and Content-Encoding.
```

That leaves:
```
refresh, content_range, accept_ranges, and www_authenticate.
```

Any reason not to add these too, so some future person generating a new header in PSOL doesn't get bitten?

**Comment by crowell** [comment]
additional shortcuts added

**Comment by jeffkaufman** [comment]
LGTM

File: src/ngx_pagespeed.cc [hunk 1]
```diff
@@ -607,6 +607,16 @@ ngx_int_t copy_response_headers_to_ngx(
       CHECK(pagespeed_headers.FindContentLength(&len));
       headers_out->content_length_n = len;
       headers_out->content_length = header;
+    } else if (STR_EQ_LITERAL(name, "Content-Encoding")) {
+      headers_out->content_encoding = header;
```
Comment (oschaaf) [inline src/ngx_pagespeed.cc]: I wonder if we should look at `r->gzip_vary` and any pre-existing `Vary: Accept-Encoding` headers here. Does PSOL also add a `Vary` header when it emits a `Content-Encoding` header?

**Pull request merged**